include/gatekeeper_mailbox.h

```diff
@@ -19,14 +19,19 @@
 #ifndef _GATEKEEPER_MAILBOX_H_
 #define _GATEKEEPER_MAILBOX_H_
 
-/*
- * TODO Remove this preprocessing directive when there are
- * are calls to these functions; for now, the mailbox is
- * just an example of how to set up a library.
- */
-#if 0
-void create_mailbox(void);
-void destroy_mailbox(void);
-#endif
+#include <rte_ring.h>
+#include <rte_mempool.h>
+
+struct mailbox {
+	struct rte_ring    *ring;
+	struct rte_mempool *pool;
+};
+
+struct mailbox *create_mailbox(
```

Comment:
Do you mean we can define the function as ``int init_mailbox(const char *tag, int ele_count, int ele_size, unsigned int lcore_id, struct mailbox *mb)``? I did not clearly see the point where we can reduce one indirection. Do you mean that we need to define the mailbox member as ``struct gk_config {xxx; struct mailbox mb;}``?